tests/test__collection_api.py

```diff
@@ -2180,6 +2180,69 @@ def test__find_or_and(self):
         ])
         self.assertEqual([2], [d['x'] for d in self.db.collection.find(search_filter)])
 
+    def test__aggregate_replace_root(self):
+        self.db.a.insert_many([
+            {'_id': 1, 'pets': {'dogs': 2, 'cats': 3}},
+            {'_id': 2, 'pets': {'hamsters': 3, 'cats': 4}}
+        ])
+        actual = self.db.a.aggregate([
+            {'$replaceRoot': {'newRoot': '$pets'}}
+        ])
+        self.assertListEqual([
+            {'dogs': 2, 'cats': 3},
+            {'hamsters': 3, 'cats': 4}
+        ], list(actual))
+
+    def test__aggregate_replace_root_non_existing(self):
+        self.db.a.insert_many([
```

Comment:
Please add a test, with the same data but with the config `'newRoot': {'dogs': '$pets.dogs'}`. The result should be:
```py
[{'dogs': 2}, {}]
```

Reply:
It seems that my code doesn't solve it, and it causes me some issues... I've added to the Missing_features document that the dot reference support is missing for replaceRoot.

Reply:
The trouble is not only in the dot:

If you add an extra data where `pets` field is missing then your code would fail. Note that you can fix (probably both by using):

```py
results = []
for doc in in_collection:
  try:
    results.append(_parse_expression(new_root, doc)
  except KeyError:
    results.appends({})
return results
```

What do you think?

Reply:
That was my first hunch so I already tested it. The problem is that we get 2 types of KeyError here: one for the case we discuss here (dot reference), and one for the case we've discussed in test test__aggregate_replace_root_non_existing.
So If I catch the KeyError here - test__aggregate_replace_root_non_existing will fail...

Reply:
You're right, I hadn't tried it and if `pets` field is missing in one doc then the query is crashing... I don't really like the current state because we're getting an OperationFailure in cases where we should get a NotImplementedError or the actual solution.

The changes would need to be larger. I think that adding a parameter to _parse_expression to ignore key errors would enable to make the difference. Let me try to do that and then you can rebase on my work.

Reply:
I've done the rebasing myself. I'll let you have a look and merge if you're OK